src/Agent.Sdk/Knob/ICompositeKnobSource.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Agent.Sdk.Knob
+{
+    public interface ICompositeKnobSource : IKnobSource
```

Comment:
Why do we need separate interface here? Could it make sense to place hasEnvironmentSourceWithName method in IKnobSource instead?

Reply:
I think this method is specific only for `CompositeKnobSource` class. Therefore we must have a separate interface. `BuiltInDefaultKnobSource` implements the `IKnobSource` interface, which, as it seems to me, cannot have such a method.